packages/core/register/src/hook.js

```diff
@@ -0,0 +1,142 @@
+import Module from 'module';
+import process from 'process';
+import path from 'path';
+import fs from 'fs';
+import {addHook} from 'pirates';
+import resolveFrom from 'resolve-from';
+
+import Parcel, {Asset, Dependency} from '@parcel/core';
+import {Cache} from '@parcel/cache';
+import syncPromise from '@parcel/utils/lib/syncPromise';
+import {loadConfig} from '@parcel/utils/lib/config';
+
+const originalRequire = Module.prototype.require;
+const DEFAULT_CLI_OPTS = {
+  watch: false
+};
+
+let hooks = {};
+let parcelDeps = [];
```

Comment:
This array keeps track of every file the hook came across that is used by Parcel.

This should prevent Parcel from processing these, but it could get quite big which is kind of an issue. Perhaps this should be an array of nearest module path or something.

Like traverse up untill it finds a package.json and add that directory, than check if file that's being checked is part of that dir and if it is skip it ...

Reply:
I wonder if it actually matters if we just process everything, including parcel and its deps. How much slower is it? Most of the time we should just be passing the code straight through with no transforming for those. We could (and probably should) even turn off finding dependencies in the code since we never use them anyway. Does it actually break things to transform everything?

Reply:
Not skipping Parcel's code makes Parcel's resolver go into an infinite recursive loop. There is also some issues with the actual transforming failing, but can't remember what the error here is. But I doubt we'll be able to get it to transform.

Performance will also be worse but it probably won't be that bad

Reply:
Ah ok.